services/ui-src/src/Routes.tsx

```diff
@@ -28,41 +28,43 @@ interface MeasureRoute {
 // eg. http://localhost:3000/:state/2021/:coreSetId/AMM-AD
 function useMeasureRoutes(): MeasureRoute[] {
   const { data } = useGetMeasureListInfo();
-  const [measureRoutes, setMeasureRoutes] = useState<MeasureRoute[]>([]);
 
-  useEffect(() => {
-    if (data) {
-      const routes: MeasureRoute[] = [];
+  const measureRoutes: MeasureRoute[] = [];
+  if (data) {
+    Object.keys(data).forEach((year: string) => {
+      data[year].forEach(
+        ({ measure, description, autocompleteOnCreation }: any) => {
+          if (measure in Measures[year]) {
+            const Comp = Measures[year][measure];
 
-      Object.keys(data).forEach((year: string) => {
-        data[year].forEach(
-          ({ measure, description, autocompleteOnCreation }: any) => {
-            if (Measures?.[year] && measure in Measures[year]) {
-              const Comp = Measures[year][measure];
-
-              routes.push({
-                key: `:state/${year}/:coreSetId/${measure}`,
-                path: `:state/${year}/:coreSetId/${measure}`,
-                element: (
-                  <QMR.MeasureWrapper
-                    name={description}
-                    year={year}
-                    measureId={measure}
-                    measure={createElement(Comp)}
-                    autocompleteOnCreation={autocompleteOnCreation ?? false}
-                  />
-                ),
-              });
-            }
+            measureRoutes.push({
+              key: `:state/${year}/:coreSetId/${measure}`,
+              path: `:state/${year}/:coreSetId/${measure}`,
+              element: (
+                <QMR.MeasureWrapper
+                  name={description}
+                  year={year}
+                  measureId={measure}
+                  measure={createElement(Comp)}
+                  autocompleteOnCreation={autocompleteOnCreation ?? false}
+                />
+              ),
+            });
           }
-        );
-      });
-
-      setMeasureRoutes(routes);
-    }
-  }, [data, setMeasureRoutes]);
+        }
+      );
+    });
+  }
 
-  return measureRoutes;
+  return data
+    ? measureRoutes
+    : [
+        {
+          key: ":state/:year/:coreSetId/:measure",
+          path: ":state/:year/:coreSetId/:measure",
+          element: <Views.MeasuresLoading />,
```

Comment:
```suggestion
          element: <QMR.LoadingWave />,
```
Is the reason you're not just doing this because of the additional CSS being added to CUI.Flex? or is there some magic about it being a view that I'm just missing?

If it is just the CSS, would it not be better just apply those styles to the LoadingWave component?

I know you're smarter than me, so I'm assuming you've already thought about this.